File: 800 - rated/UnitArray.java
import java.util.Scanner;
public class UnitArray{
    public static void main(String[] args){
        Scanner sc = new Scanner(System.in);
        int t=sc.nextInt();
        while(t-- != 0){
            int n = sc.nextInt();
            int pos=0,neg=0;
            int temp;
            int result = 0 ;
            for(int i=0;i<n;i++){
                temp = sc.nextInt();
                if(temp < 0) neg++;
                
            }
            pos = n -neg;
            if(neg > pos){
                int count = (neg - pos + 1)/2;
                result += count;
                neg -= count ;
                pos += count;



            }
            if(neg % 2 == 1 ) result += 1;
            System.out.println(result);
        }
    }
}

File: 800 - rated/unit_array.py
t = int(input())
for _ in range(t):
    n = int(input())
    nums = map(int,input().split())
    neg = 0
    pos = 0
    k = 0
    for i in nums: 
        if i > 0:
            pos+=1
        else:
            neg+=1
    while True :
        if neg % 2 == 0 and pos - neg >= 0 :
            print(k)
            break
        if neg % 2 == 1 :
            k+=1
            neg-=1
            pos+=1
        if pos - neg < 0 and neg % 2 == 0:
            neg -=2
            pos +=2
            k+=2
        